### DISCUSSION TITLE
not using absolute path?

### DISCUSSION THREAD

**Discussion by codyng** [vercel/next.js#12058] [Help]
Hi there, I know next.js is using absolute path after build (or export).

Although I know we can config the prefix with the next config file, is it possible to do without using absolute path, so that they request the js or css within the same folder?

It is because, sometimes we just build a site that is deployed to a big site under one of the sub-path, and the files are bounded within the folder so we can't upload the _next folder outside it.

Just wondering if it supports disabling using absolute path build in future. Thanks.

**Comment by timneutkens** [comment]
> Just wondering if it supports disabling using absolute path build in future. Thanks.

I'm not entirely sure what you mean by this? The assetPrefix option allows you to change the location: https://nextjs.org/docs/api-reference/next.config.js/cdn-support-with-asset-prefix

**Reply by codyng** [reply]
Oh... sorry, may be I should explain better. After exporting the site, the URL paths inside the HTML files are in absolute path (started with /, i.e. /_next/static/xxxx). My question is whether it is possible to use relative path, i.e. _next/static/xxxx.

The advantage is that we can deploy the site under any sub-directory, without "hard pre-defined" the assetPrefix per each export. Most of the time, we don't know whether the end users (e.g. clients) will deploy the site whether under the base path / or under other sub-paths. They can move or deploy the next.js site easily.

**Reply by timneutkens** [reply]
That'll break too as we inject additional script tags even when you move between pages, hence why you have to set `assetPrefix` to a separate location in you want to do what you're proposing.
